lib/internal/validators.js

```diff
@@ -113,6 +113,19 @@ const validateUint32 = hideStackFrames((value, name, positive) => {
   }
 });
 
+const validateUint32OrNegative1 = hideStackFrames((value, name) => {
+  if (!isUint32(value) && value !== -1) {
+    if (typeof value !== 'number') {
+      throw new ERR_INVALID_ARG_TYPE(name, 'number', value);
+    }
+    if (!NumberIsInteger(value)) {
+      throw new ERR_OUT_OF_RANGE(name, 'an integer', value);
+    }
+    // 2 ** 32 === 4294967296
+    throw new ERR_OUT_OF_RANGE(name, '>= -1 && < 4294967296', value);
+  }
+});
```

Comment:
Instead of using a completely new validation function, I would just change the `positive` argument in `validateUint32` to become `minimum`. That way the function is able to handle all kinds of minimums, not only zero and one.

Reply:
Allowing an arbitrary minimum doesn’t make sense for a function called `validateUint32`, though.

Reply:
It's okay, `validateInt32` has `min` and `max` parameter, too

Reply:
I agree with @addaleax, the fact that `validateInt32` have these parameters doesn't mean that they make sense for `validateUint32`, it is not possible for `Uint` to be negative, by the definition of that type (hence the failing `isUint32(value)` check), therefore it would be strange to have it support such a use case.
I think it would be better to just check this manually (the same way we do for optional values by just checking `if (value !== undefined)` before validation) with 
```js
if (uid !== -1) validateUint32(...)
```

If we want to generalize this, I think `validateInteger` would be a better fit, by adding `min`/`max` to it.